test/features/steps/steps.py

```diff
@@ -16,55 +17,54 @@
 if not hasattr(base64, 'encodestring'):
     base64.encodestring = base64.encodebytes
 
+# Monkey-patch Behaving to handle function rename
+from behaving.web.steps import forms
+if not hasattr(forms, 'fill_in_elem_by_name'):
+    forms.fill_in_elem_by_name = forms.i_fill_in_field
+
 URL_RE = re.compile(r'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|\
                     (?:%[0-9a-fA-F][0-9a-fA-F]))+', re.I | re.S | re.U)
 
 
-@step(u'I get the current URL')
-def get_current_url(context):
-    context.browser.evaluate_script("document.documentElement.clientWidth")
-
-
-@step(u'I go to homepage')
+@when(u'I go to homepage')
 def go_to_home(context):
     when_i_visit_url(context, '/')
 
 
-@step(u'I go to register page')
+@when(u'I go to register page')
 def go_to_register_page(context):
     context.execute_steps(u"""
         When I go to homepage
         And I click the link with text that contains "Register"
     """)
 
 
-@step(u'I log in')
+@when(u'I log in')
 def log_in(context):
-    assert context.persona
     context.execute_steps(u"""
         When I go to homepage
-        And I resize the browser to 1024x2048
+        And I maximize the browser's window
         And I click the link with text that contains "Log in"
         And I log in directly
     """)
 
 
-@step(u'I log in directly')
+@when(u'I log in directly')
 def log_in_directly(context):
     """
     This differs to the `log_in` function above by logging in directly to a page where the user login form is presented
     :param context:
     :return:
     """
 
-    assert context.persona
+    assert context.persona, "A persona is required to log in, found [{}] in context. Have you configured the personas in before_scenario?".format(context.persona)
```

Comment:
so a nice error message?

Reply:
Basically. Due to a lesson learned.

The fact that Behaving wipes all personas before each scenario is annoying, but what can you do.